isisd/isis_zebra.c

```diff
@@ -593,7 +651,11 @@ void isis_zebra_init(struct thread_master *master, int instance)
 	zclient_sync = zclient_new(master, &zclient_options_default);
 	zclient_sync->sock = -1;
 	zclient_sync->redist_default = ZEBRA_ROUTE_ISIS;
-	zclient_sync->instance = instance;
+	/*
+	 * Instance must be different in order for zclient to distinguish
+	 * the asynchronous socket from the synchronous one
+	 */
+	zclient_sync->instance = instance + 1;
```

Comment:
This is actually not needed - the caller of the function is passing 1 as the instance [(here)](https://github.com/FRRouting/frr/blob/dev_isis_sr/isisd/isis_main.c#L201) and the asynchronous client is not initializing the field in its struct, meaning that it uses 0. I would suggest to either remove this line and leave everything as is, or also initializing the asynchronous client to use "instance" so that they have respectively values `instance` (for the async client) and `instance+1` (for the sync)

Reply:
@Renato Good catch. I forgot this part, focusing only on the Hello Message.
@manuhalo I incorporate the `instance` parameter for the asynchronous socket. But, as initial `instance` value is set to`1` in isisd/isis_main.c, I change to `instance - 1` for the asynchronous socket and `instance` for the synchronous socket.

Reply:
Since this issue is already being taken care of in other PRs, please drop this change and this PR should be ready to go.